buildSrc/src/main/kotlin/tasks/InstallAndroidSdkTask.kt

```diff
@@ -0,0 +1,94 @@
+package tasks
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.UntrackedTask
+import org.gradle.tooling.GradleConnector
+import java.io.File
+
+/**
+ * Installs Android SDK. This task requires the ANDROID_SDK_ROOT env variable to be set.
+ */
+@UntrackedTask(because = "Output directory can change when running other builds")
+abstract class InstallAndroidSdkTask : DefaultTask() {
+
+    @get:Input
+    abstract val androidSdkVersion: Property<String>
+
+    @get:OutputDirectory
+    abstract val androidProjectDir: DirectoryProperty
+
+    @TaskAction
+    fun install() {
+        if (System.getenv("ANDROID_SDK_ROOT") == null) {
+            throw GradleException(
+                "ANDROID_SDK_ROOT env variable is not set but it should be with the installation directory path. " +
+                    "The installation directory should also contain the Android sdk license key. See https://developer.android.com/studio/intro/update.html#download-with-gradle."
+            )
+        }
+
+        val androidSdkVersion = androidSdkVersion.get()
+        val projectDir = androidProjectDir.get().asFile
+
+        createAndroidProject(projectDir, androidSdkVersion)
+        buildAndroidProject(projectDir)
+    }
+
+    private fun buildAndroidProject(projectDir: File) {
+        try {
+            val connector = GradleConnector.newConnector()
+                .forProjectDirectory(projectDir)
+            connector.connect().use {
+                it.newBuild().forTasks("build").run()
+            }
+        } catch (e: Exception) {
+            e.printStackTrace()
```

Comment:
Something like that?
```
            it.newBuild().forTasks("build")
                .addArguments("--info")
                .setStandardError(System.err)
                .setStandardOutput(System.out)
                .run()```

Reply:
That would always print the output, right? I am not sure how useful this would be. I'd like to print the output only when the task fails.

I don't think `--info` is necessary.